### PR TITLE
BC-6849 - Implement BoardStore with Pinia

### PR DESCRIPTION
**Pull request by muratmerdoglu-dp**
# Short Description

## Links to Ticket and related Pull-Requests

https://ticketsystem.dbildungscloud.de/browse/BC-6849

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team
- [x] DEV: Every new component is implemented having accessibility in mind (e.g. aria-label, role property)

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/components/data-board/store/BoardStore.ts [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+		updateColumnTitleCall,
+		updateBoardVisibilityCall,
+		createCardCall,
+	} = useBoardApi();
+
+	const { setEditModeId } = useSharedEditMode();
+
+	const { handleError, notifyWithTemplate } = useErrorHandler();
+
+	function dispatch(action: PermittedStoreActions<typeof BoardActions>) {
```
Comment (uidp) [inline src/components/data-board/store/BoardStore.ts]: I don't really get the whole point of `dispatch` and the `ActionFactory`, sorry. Why can't we just call the Pinia actions directly like `createCard(columnId)`? The typescript compiler would then make sure that the store implements the action, right? No need for `PermittedActions`. Or am I missing something here?

File: src/modules/data/board/store/BoardStore.ts [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+			on(BoardActions.deleteColumn, deleteColumn),
+			on(BoardActions.moveCard, moveCard),
+			on(BoardActions.moveColumn, moveColumn),
+			on(BoardActions.updateColumnTitle, updateColumnTitle),
+			on(BoardActions.updateBoardTitle, updateBoardTitle),
+			on(BoardActions.updateBoardVisibility, updateBoardVisibility),
+			on(BoardActions.updateBoardVisibility, updateBoardVisibility),
+			on(BoardActions.reloadBoard, reloadBoard)
+		);
+	}
```
Comment (uidp) [inline src/modules/data/board/store/BoardStore.ts:54]: I don't really get the whole point of dispatch and the ActionFactory, sorry. Why can't we just call the Pinia actions directly like createCard(columnId)? The typescript compiler would then make sure that the store implements the action, right? No need for PermittedActions. Or am I missing something here?

**Review by hoeppner-dataport — APPROVED** [review]

**Review by hoeppner-dataport — DISMISSED** [review]

**Review by hoeppner-dataport — APPROVED** [review]

File: src/modules/data/board/BoardStore.ts [hunk 3]
```diff
@@ -26,6 +28,25 @@ export const useBoardState = (id: string) => {
...
+			on(BoardActions.createCard, createCard),
+			on(BoardActions.createColumn, createColumn),
+			on(BoardActions.deleteCard, deleteCard),
+			on(BoardActions.deleteColumn, deleteColumn),
+			on(BoardActions.moveCard, moveCard),
+			on(BoardActions.moveColumn, moveColumn),
+			on(BoardActions.updateColumnTitle, updateColumnTitle),
+			on(BoardActions.updateBoardTitle, updateBoardTitle),
+			on(BoardActions.updateBoardVisibility, updateBoardVisibility),
+			on(BoardActions.updateBoardVisibility, updateBoardVisibility),
```
Comment (virgilchiriac) [inline src/modules/data/board/BoardStore.ts:45]: duplicate line

Comment (hoeppner-dataport) [reply src/modules/data/board/BoardStore.ts:45]: true - and it sets to visible and removes it again.

**Pull request merged**